core/domain/exp_services.py

```diff
@@ -1998,28 +2134,93 @@ def update_exploration(
     updated_exploration = apply_change_list(exploration_id, change_list)
     if get_story_id_linked_to_exploration(exploration_id) is not None:
         validate_exploration_for_story(updated_exploration, True)
-    _save_exploration(
-        committer_id, updated_exploration, commit_message, change_list)
-
-    discard_draft(exploration_id, committer_id)
-
-    # Update summary of changed exploration in a deferred task.
-    taskqueue_services.defer(
-        taskqueue_services.FUNCTION_ID_REGENERATE_EXPLORATION_SUMMARY,
-        taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS, exploration_id,
-        committer_id)
+    exploration_rights = rights_manager.get_exploration_rights(
+        updated_exploration.id
+    )
+    exploration_is_public = (
+        exploration_rights.status != rights_domain.ACTIVITY_STATUS_PRIVATE
+    )
+    updated_exploration.validate(strict=exploration_is_public)
+    models_to_put.extend(
+        _compute_models_to_put(
```

Comment:
Hmm -- when you rename this (as suggested above), make sure it's clearly different semantically than the name of the current method, which has a similar idea. The name should clearly identify which subset of the "compute model" functionality that function is responsible for handling.